cpp/src/arrow/io/interfaces.h

```diff
@@ -262,8 +262,8 @@ class ARROW_EXPORT RandomAccessFile : public InputStream, public Seekable {
   /// \param[in] file_offset the starting position in the file
   /// \param[in] nbytes the extent of bytes to read. The file should have
   /// sufficient bytes available
-  static std::shared_ptr<InputStream> GetStream(std::shared_ptr<RandomAccessFile> file,
-                                                int64_t file_offset, int64_t nbytes);
+  static Result<std::shared_ptr<InputStream>> GetStream(
+      std::shared_ptr<RandomAccessFile> file, int64_t file_offset, int64_t nbytes);
```

Comment:
@pitrou should we keep the original non-Result signature as well for backwards compatibility?

Reply:
It's not possible: you cannot have two overloads with the same exact arguments.